protocols/floodsub/src/protocol.rs

```diff
@@ -58,8 +86,10 @@ where
 
     fn upgrade_inbound(self, mut socket: TSocket, _: Self::Info) -> Self::Future {
         Box::pin(async move {
-            let packet = upgrade::read_length_prefixed(&mut socket, 2048).await?;
-            let rpc = rpc_proto::Rpc::decode(&packet[..]).map_err(DecodeError)?;
+            let packet = upgrade::read_length_prefixed(&mut socket, MAX_MESSAGE_LEN_BYTES).await?;
+            // Replace with prost_codec::Codec
+            let rpc = self.codec.decode();
```

Comment:
@thomaseizinger (When you've got a minute, I've got a rust question that I can't seem to find an answer for)
 
The compiler complains about this line and it doesn't make sense to me. The clippy error is:

```
    |
91  |             let rpc = self.codec.decode();
    |                                  ^^^^^^ this is an associated function, not a method
    |
```
It offers suggestions to "fix" this, but those suggestions just lead to more errors/complaints.

My confusion: `self` is identified as a `FloodsubProtocol` object. `codec` is a field there, and the `new()` is defined and sets `codec`... so it seems to me like there's an initialized field there.

What have I missed?

Reply:
What you probably want to do here is use a `Framed` to wrap the incoming socket and immediately read a message from it using its `Stream::next` implementation.

`Framed` is an abstraction on top of a codec to read and write from a socket in entire messages instead of raw bytes!

Reply:
This is awesome, thank you! I'm still working my way through it, work has kept me busier-than-usual this week.